WcaOnRails/app/models/competition.rb

```diff
@@ -27,7 +27,7 @@ class Competition < ApplicationRecord
            allow_nil: true,
            with_model_currency: :currency_code
 
-  scope :not_over, -> { where("end_date >= ?", Date.today) }
+  scope :not_over, -> { where("end_date >= ? AND results_posted_at IS NULL", Date.today) }
```

Comment:
OK. The I'm out of ideas. You can see the separate methods as a speed optimization.